src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -406,9 +399,19 @@ public void close() {
             resubscribeFuture.cancel(true);
         }
         try {
-            disconnect().get(getCloseTimeout(), TimeUnit.MILLISECONDS);
+            disconnect().get(getTimeout(), TimeUnit.MILLISECONDS);
         } catch (InterruptedException | ExecutionException | TimeoutException e) {
             logger.atError().log("Error while disconnecting the MQTT client", e);
         }
     }
+
+    public synchronized void closeOnShutdown() {
```

Comment:
I don't think this should be synchronized. You need to sync very carefully in this class because we've had deadlocks before.

Reply:
will update and suppress FB warnings instead.